### ISSUE TITLE
Request?  Possible to pull tokens from compendiums?

### ISSUE THREAD

**Issue by kenpoh01** [spetzel2020/quick-encounters#14]
Hello,
I'm wondering if it would be possible to have the module save the position of the tokens, which could be found in a compendium (not in the actors directory) and still be able to both import them from the compendium, then place them in the scene?  I'm probably asking a lot here.  

My reasoning is that I don't have all of my tokens in the actors directory as I think that slows down load times.

**Comment by spetzel2020** [comment]
I completely agree. Compendium entries would be consistent with not importing/placing tons of actors on a busy map.

Ironically I have already put this into 0.6, but didn't have time to test all the interactions and make sure it worked properly so i didn't include it in the description of the release.

Right now, you can drag a Compendium entry directly into the Journal, and Quick Encounters will recognize it as a new Quick Encounter. Unfortunately when you click "Run Quick Encounter" it will import the entry but doesn't place the the token or run the Encounter. 

I don't think those are difficult problems to fix, and plan to do them as part of a maintenance version.

However, a more difficult problem is that Foundry won't let you place a Compendium on the map without automatically importing the Actor. So you'd be restricted to Method 2, where the tokens are clustered randomly around the Map Note. This is probably fine for a first version.

I'd love to get your feedback.

**Comment by kenpoh01** [comment]
Thanks for the response,

Well, if you want to see an honest reaction to your module, check out the video I made yesterday on my first install and trial.  https://youtu.be/QJcP5J8lEhQ

**Comment by spetzel2020** [comment]
Hey, I watched your video and it was excellent feedback. It highlighted a couple of places where I could make it clearer what to do. And of course the Compendium ability :)

fyi, if you just select tokens without having an open Journal Entry, QE will create the new Journal Entry for you. Or you can do it the way you did - makes no difference except the QE Journal Entry has more verbiage in it explaining what's happening.

You can also tap the QE button with nothing selected and it'll give you a Tutorial.

Anyway, tl;dr: really appreciate your feedback and comments - it's the best way to improve something!

**Comment by spetzel2020** [comment]
I'm think again about your video and want to make sure I saw what I think I did.

I saw you drag a Compendium entry to the Scene and Foundry automatically imported the Actor. Then you created the Quick Encounter and then **deleted the Actor**. 

I can see how you might **want** to do that to save on Actors (to only import them on demand) but it's a very difficult use case because after you import the Actor, Foundry doesn't track which Compendium it came from. So Quick Encounters has no way of knowing anything other than you deleted the Actor it was expecting to see.

What I have coded is that you can drag Compendium entries into Journal Entries and then Quick Encounters will automatically create a Quick Encounter, albeit not one with saved tokens, but one where the tokens are placed around the Map Note. THis is still buggy, but I can probably get that working if it addresses the need.

**Comment by kenpoh01** [comment]
I think I could make that work too.  If Quick Encounter can load them in, then place them on the scene, that still saves from having them in the scene to begin with.  If it were possible to load them in "hidden" so that they could be placed where they need to go, then they could be "unhidden" to start the combat.

  I am trying to save on Actors filling my Actors directory, but if they can't be pulled from the compendiums, that's fine.

**Comment by spetzel2020** [comment]
As you say, when they weren't saved as tokens originally, they come in hidden by default. You can override the behavior for saved or generated tokens by Ctrl-Run (makes everything visible for shock value) or Alt-Run (makes everything invisible).

As you can probably gather from the new QE dialog, my direction is to avoid your having to edit the Journal Entry to change encounters, but hopefully allowing you to. It's tricky. For the moment, I'll fix the Compendium bug and then see if I can enable dragging a Compendium entry directly into the QE dialog.

**Comment by spetzel2020** [comment]
Compendium Actors in Journal Entries are now working in v0.6.11 (latest release). However to use them, you reference them in the Journal Entry and they will create an appropriate entry in the QE dialog.

There is a minor bug: the COmpendium image doesn't always load quickly (an issue with lazy loading of images and race condition) but the major remaining problem is that you can't positions tokens - they can only be generated (and they'll appear near the Scene Note).

For v0.7 I'm planning on over-hauling token placement and hopefully allowing you to edit the placement - I'll try to build in support for Compendium token placement as well.

**Comment by kenpoh01** [comment]
Hmmm.  Can't seem to get it to work from compendium.  Am I doing it right?  I create a journal entry, drag creatures from the compendium into the journal entry, then, try to run the encounter?  It asks to put a note in the scene, which it does, I see the monsters appear in my Actors side bar, but no actors get added to the scene.
![image](https://user-images.githubusercontent.com/36082436/100958766-f148be00-34ea-11eb-83cf-f2f0e5a78895.png)

**Comment by spetzel2020** [comment]
Hmm, that's not good. Can you check a couple of things:
- Verify that you are on 0.6.11 ( you should be able to check my going to Manage Modules)
- Can you open the browser console (F12 in Windows Chrome) and see if there are any errors in the log?
- Also, can you export the Journal Entry and DM me the JSON? Spetzel#0103 on Discord

**Comment by kenpoh01** [comment]
So, just checked and I'm still on version 0.6.5 but it won't update automatically to 0.6.11 (says current).

**Comment by spetzel2020** [comment]
That's concerning that it wouldn't auto-update. I'll have to check into it.
Glad to hear that the Compendium is working better!

**Comment by spetzel2020** [comment]
I'm going to close this Bug in 0.6.11/12 and open an Enhancement to allow you to position Tokens from the Compendium

**Comment by spetzel2020** [comment]
Enhancement request created: https://github.com/spetzel2020/quick-encounters/issues/19
This Issue closed as a fixed bug in 0.6.12
